homeassistant/components/lektrico/sensor.py

```diff
@@ -0,0 +1,314 @@
+from homeassistant.const import (
+    CONF_FRIENDLY_NAME,
+    PERCENTAGE,
+    UnitOfElectricCurrent,
+    UnitOfElectricPotential,
+    UnitOfEnergy,
+    UnitOfPower,
+    UnitOfTemperature,
+    UnitOfTime,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import LektricoDeviceDataUpdateCoordinator
+from .const import DOMAIN
+
+
+@dataclass
+class LektricoSensorEntityDescription(SensorEntityDescription):
+    """A class that describes the Lektrico sensor entities."""
+
+    value: Callable[[Any], float | str | int] | None = None
+
+
+SENSORS_FOR_CHARGERS: tuple[LektricoSensorEntityDescription, ...] = (
+    LektricoSensorEntityDescription(
+        key="state",
+        name="State",
+        value=lambda data: str(data.charger_state),
+    ),
+    LektricoSensorEntityDescription(
+        key="charging_time",
+        name="Charging time",
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        value=lambda data: int(data.charging_time),
+    ),
+    LektricoSensorEntityDescription(
+        key="power",
+        name="Power",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.POWER,
+        native_unit_of_measurement=UnitOfPower.KILO_WATT,
+        value=lambda data: float(data.instant_power) / 1000,
+    ),
+    LektricoSensorEntityDescription(
+        key="energy",
+        name="Energy",
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        value=lambda data: float(data.session_energy) / 1000,
+    ),
+    LektricoSensorEntityDescription(
+        key="temperature",
+        name="Temperature",
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.CELSIUS,
+        value=lambda data: float(data.temperature),
+    ),
+    LektricoSensorEntityDescription(
+        key="lifetime_energy",
+        name="Lifetime energy",
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        device_class=SensorDeviceClass.ENERGY,
+        native_unit_of_measurement=UnitOfEnergy.KILO_WATT_HOUR,
+        value=lambda data: int(data.total_charged_energy),
+    ),
+    LektricoSensorEntityDescription(
+        key="installation_current",
+        name="Installation current",
+        device_class=SensorDeviceClass.CURRENT,
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        value=lambda data: int(data.install_current),
+    ),
+    LektricoSensorEntityDescription(
+        key="limit_reason",
+        name="Limit reason",
+        value=lambda data: str(data.current_limit_reason),
+    ),
+)
+
+SENSORS_FOR_LB_DEVICES: tuple[LektricoSensorEntityDescription, ...] = (
```

Comment:
Instead of using separate lists, it may be worth considering adding a  supported lambda, then evaluate that when adding the entities below. I don't think theres anything wrong with this as it is, but that is how a lot of the integrations do it.